### ISSUE TITLE
AE2FC Universal Wireless Terminal Causing significant delays

### ISSUE THREAD

**Issue by JodieRuth** [GTNewHorizons/GT-New-Horizons-Modpack#14119]
### Your GTNH Discord Username

JodieRuth

### Your Pack Version

2.3.5

### Your Server

SP

### Java Version

Java 8

### Type of Server

Vanilla Forge

### Your Expectation

Open the Universal Wireless Terminal to interact with the ME network.

### The Reality

Open it as follows. Performing any operation on this interface, including retrieving and storing items, requesting automatic crafting, or checking the crafting status, can cause significant delays (not only on the client side). It looks like the game has been frozen. Turning off Universal Wireless Terminal has no effect, and requests for automatic crafting calculations can sometimes result in the game not responding. This situation does not occur in any wireless terminals of AE2 Wireless Crafting Terminal or terminals that rely on ME Cable, and all wireless terminals of AE2FC have this issue.This is even more evident on the massive ME network at the end of the game.
![2XV{YISIC_)9DYS X7E@V_7](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/132600151/df531705-cc71-43e6-a843-0fe1cbb3b62c)
打开后如下。在这个界面进行任何操作，包括拿取与存入物品，请求自动crafting或者检查Crafting status均会造成非常大的延迟（不止存在于客户端）。看起来就像游戏被冻结了。关闭Universal Wireless Terminal无作用，请求自动crafting计算时有时会导致游戏未响应。这种情况未出现在AE2 Wireless Crafting Terminal的任意Wireless Terminal或者需要依赖ME Cable的Terminal，AE2FC的所有Wireless Terminal均有此问题。这在游戏末期的极大规模的ME网络上更为明显。

### Your Proposal

AE2FC Universal Wireless Terminal is a very useful tool, please fix it

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [X] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by firenoo** [comment]
As a temporary workaround, the server owner can set a few parameters in the ae2fc config:
* `#packetRate` is the number of milliseconds before a packet can be sent (higher means less lag), but also makes the terminal less responsive to player interactions
* I am not sure what effect the other parameter (packet size) have to be honest, just that it increases number of items sent over the network in 1 packet

**Comment by JodieRuth** [comment]
> As a temporary workaround, the server owner can set a few parameters in the ae2fc config:
> 
> * `#packetRate` is the number of milliseconds before a packet can be sent (higher means less lag), but also makes the terminal less responsive to player interactions
> * I am not sure what effect the other parameter (packet size) have to be honest, just that it increases number of items sent over the network in 1 packet

Yes, this works, unfortunately it makes Universal Wireless Terminal appear to be running in an environment with extremely high latency, but at least for now it will no longer freeze the server. Still not a perfect solution.

**Comment by boubou19** [comment]
does it still happen on 2.3.6?

**Comment by Ilithios06** [comment]
I can confirm that this issue still prevails on 2.3.6.
This is the Network analysis in the task manager, you can clearly see when terminals are opened by the extreme spikes. Hence doing many operations with terminals such as encoding, requesting, inserting patterns in a pattern terminal in rapid succession cause severe lag. The issue also seems to appear with all types of terminals, wireless or not, at least thats my experience.
![image](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/86689318/98178965-5e2b-4e36-a400-ff13f18c9314)

**Comment by JodieRuth** [comment]
Only the AE2FC wireless terminal has this issue, and everything is normal between AE2 and another MOD (I forgot its name) wireless terminal. 2.3.6 There is still this issue, and modifying cfg will only make wireless terminals work like in an environment with significant lag. This issue does not seem to occur on locally launched servers and single player games.

**Comment by zonkedzoey** [comment]
Confirmed in 2.5, seems to be the same issue as https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/14495

Renders use on my dedicated server not possible as it saturates the network

**Comment by Eysla** [comment]
Still an issue on 2.6.1, making the universal terminal completely unusable for anyone with poor download speeds playing on dedicated servers.

**Comment by Ruling-0** [comment]
@Eysla @zonkedzoey @JodieRuth are any of you able to download a recent nightly version (https://gtnh.miraheze.org/wiki/Nightly_Builds) and confirm if this still occurs in 2.7?

**Comment by duckedddd** [comment]
Hi yes i also experience this on servers, got around it by using exitlag. From OCE and can play on EU /US servers no problem now, but there is still def something up with ae2fc terminals. I know multiple people who experience this issue particularly with larger ae networks.

**Comment by PLASMAchicken** [comment]
### Your GTNH Discord Username

_No response_

### Your Pack Version

2.8.1

### Your Proposal

When the Wireless Universal Terminal (`ae2fc:wireless_ultra_terminal`) is opened on a server, it sustains high network bandwidth usage (~500K/s) for extended periods—even while idle. This constant traffic places unnecessary strain on server bandwidth. Video demonstration: [Watch here](https://github.com/Huaxidesu/Picture-of-huaxidesu/raw/refs/heads/%E4%B8%BB%E8%A6%81/GTNH/%E6%97%A0%E7%BA%BF%E7%BB%88%E7%AB%AF%E5%BC%82%E5%B8%B8%E5%8D%A0%E7%94%A8.mp4). Please optimize the terminal to avoid persistent high-bandwidth transmission—data should only sync when necessary.

### Your Goal

This optimization is needed because the sustained bandwidth consumption from the Wireless Universal Terminal significantly impacts server performance and increases operational costs, especially in multiplayer environments with limited bandwidth or many concurrent players. Reducing this persistent network load will improve server stability and scalability without affecting usability.

### Your Vision

I envision the Wireless Universal Terminal transmitting data only on demand—for example, when the player interacts with it or when inventory contents change—rather than continuously streaming updates. This would eliminate the unnecessary sustained network traffic, making the mod far more server-friendly while preserving full functionality.

### Final Checklist

- [x] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this change please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [x] I understand this change request may not attract enough attention and thus not be implemented.
- [x] I understand this change request may be rejected due to other community members think it's inappropriate.
- [x] I believe this feature would make the pack better.

**Comment by mgomezch** [comment]
I just hit this on 2.8.1, then I upgraded to 2.8.4 and it still happens.  I can reproduce it 100% reliably.  I'm not sure it wasn't happening before, maybe it was but to a lesser extent and my bandwidth could tolerate it, but now it makes the terminal entirely unusable; having it open for even one minute makes the ping stack so high that it basically never recovers.  I tried tweaking the settings mentioned above but it didn't help; I don't know what values can help with this.  Any idea how to work around this?  I'm surprised it's not more common, I'm only in early LuV so my ME system probably doesn't have anywhere close to as many types to a typical midgame/lategame player.  It's very weird.  It worked fine for at least 100 hours and started after seemingly no particular change other than playing normally.

**Comment by mgomezch** [comment]
I think I figured out what triggers it for me.  If I put my collection of EnderIO soul vials (with mobs in them) somewhere that AE2 can see it, then this reproduces reliably; if I hide them from AE2 by taking them out (either into an inventory that's not connected to AE2 at all, or in a barrel with an insert-only storage bus fuzzy-filtered to soul vials with mobs in them and I flip item visibility to only extractable (so they wouldn't be listed in the terminal since the storage bus is insert-only), then the bug goes away without changing anything else, confirmed with the Opis ping graph.  I'm not sure if it's the number of soul vials, or a specific mob in one of them with weird data.  I'll try to bisect the vials I guess.

**Comment by mgomezch** [comment]
After some more testing, while removing my soul vials from AE2 definitely made the problem more manageable, this does keep happening even without them.  I guess they were just tipping this code over the edge with too much extra data that triggered whatever causes this, but my game still slows down a lot even now.

**Comment by Worive** [comment]
Should have been fixed https://github.com/GTNewHorizons/AE2FluidCraft-Rework/pull/340 (merged after 2.8.4), so any beta 2.9 should already have it
